### DISCUSSION TITLE
Using SelectedIndex on Combobox does not show selected item

### DISCUSSION THREAD

**Discussion by FraserElectronics** [AvaloniaUI/Avalonia#13288] [Q&A]
## Describe the bug
I populate a ComboBox at runtime and once I have added all my items, I set the SelectedIndex property of the Combobox via an ObservableProperty. However, the combox remains blank and does not show the selected item.

## To Reproduce

This is my very simple test view:

```
	<StackPanel Orientation="Vertical" Margin="8">
		<ComboBox Grid.Row="2" Grid.Column="0" ItemsSource="{Binding Items}" SelectedIndex="{Binding SelectedItemIndex}" Width="300"/>
		<TextBlock Grid.Row="3" Text="{Binding SelectedItemText}"/>
		<Button Grid.Row="5" Command="{Binding Populate}">
			Populate Items
		</Button>
	</StackPanel>
```

And the view model that is attached to this view:

```
public partial class MainViewModel : ViewModelBase
{
    [ObservableProperty]
    private ObservableCollection<string> items = new ObservableCollection<string>();

    [ObservableProperty]
    private int selectedItemIndex;

    [ObservableProperty]
    private string selectedItemText;

    [RelayCommand]
    public void Populate()
    {
        Items.Clear();

        for ( int i = 1 ; i < 11 ; i++ )
        {
            Items.Add( $"Item {i}" );
        }

        SelectedItemIndex = 0;
    }

    public MainViewModel()
    {
        SelectedItemText = "NO SELECTION";
    }

    partial void OnSelectedItemIndexChanged( int value )
    {
        SelectedItemText = Items[ value ];
    }
}
```

## Expected behavior
I would expect to see the first item text to appear in the Combobox rather than it remaining blank as shown below.

## Screenshots

![image](https://github.com/AvaloniaUI/Avalonia/assets/19693425/b5915a45-0cd0-424d-8386-cd1703ec2a56)

## Environment 

- Windows 11 x64
- Avalonia 11.0.4

Andy

**Comment by FraserElectronics** [comment]
Okay, it seems that I need to use the SelectedItem to cause the selected item to appear.
I would say that this is a bug - specifying SelectedIndex should also show the selected item.

Andy

**Comment by timunie** [comment]
okay that's not an actual bug. SelectedIndex is `0` by default. So if you set it to `0` again, the UI does not get notified as nothing has changed for the ViewModel. Either set `SelectedIndex=-1` or `RaisePropertyChanged(nameof(SelectedIndex))` in populate. See:

```diff
public partial class MainViewModel : ViewModelBase
{
    [ObservableProperty]
    private ObservableCollection<string> items = new ObservableCollection<string>();

    [ObservableProperty]
!   private int selectedItemIndex = -1;

    [ObservableProperty]
    private string selectedItemText;

    [RelayCommand]
    public void Populate()
    {
        Items.Clear();

        for ( int i = 1 ; i < 11 ; i++ )
        {
            Items.Add( $"Item {i}" );
        }

        SelectedItemIndex = 0;
    }

    public MainViewModel()
    {
        SelectedItemText = "NO SELECTION";
    }

    partial void OnSelectedItemIndexChanged( int value )
    {
        SelectedItemText = Items[ value ];
    }
}
```

> **NOTE**
> I'll make a discussion from it. Please ask questions not as an issue but as a Q&A or on Telegram if you're new to all of this. We can still make an issue of it if it turns out it acutally is an issue. Thx. 🙏

**Reply by FraserElectronics** [reply]
@timunie  Okay, got it 👍

Thanks.

Andy

**Comment by jpgarza93** [comment]
I still believe there is a bug with combobox. I am seeing the same issue with SelectedIndex. If I bind to SelectedIndex when dynamically populating ItemSource during runtime, it won't allow me to select any of the available items. It shows me all items, it just wont register a selection.

**Reply by jp2masa** [reply]
Related: #13736
